timeline-chart/src/layer/time-graph-range-events-layer.ts

```diff
@@ -1,27 +1,27 @@
 import { TimeGraphRectangle } from "../components/time-graph-rectangle";
 import { TimelineChart } from "../time-graph-model";
 import { TimeGraphChartProviders } from "./time-graph-chart";
-import { TimeGraphLayer } from "./time-graph-layer";
+import { TimeGraphViewportLayer } from "./time-graph-viewport-layer";
 
-export class TimeGraphRangeEventsLayer extends TimeGraphLayer {
+export class TimeGraphRangeEventsLayer extends TimeGraphViewportLayer {
     protected rangeEvents: Map<TimelineChart.TimeGraphAnnotation, TimeGraphRectangle>;
     protected providers: TimeGraphChartProviders;
 
-    private _viewRangeUpdateHandler: { (): void; (viewRange: TimelineChart.TimeGraphRange): void; (viewRange: TimelineChart.TimeGraphRange): void; };
+    private _worldRangeUpdateHandler: { (): void; (worldRange: TimelineChart.TimeGraphRange): void; (worldRange: TimelineChart.TimeGraphRange): void; };
 
     constructor(id: string, providers: TimeGraphChartProviders) {
         super(id);
         this.providers = providers;
     }
 
     protected afterAddToContainer() {
-        this._viewRangeUpdateHandler = (): void => this.update();
-        this.unitController.onViewRangeChanged(this._viewRangeUpdateHandler);
+        this._worldRangeUpdateHandler = (): void => this.update();
+        this.viewport.onWorldRender(this._worldRangeUpdateHandler);
     }
 
     protected addRangeEvent(rangeEvent: TimelineChart.TimeGraphAnnotation) {
-        const start = this.getPixel(rangeEvent.range.start - this.unitController.viewRange.start);
-        const end = this.getPixel(rangeEvent.range.end - this.unitController.viewRange.start);
+        const start = this.getWorldPixel(rangeEvent.range.start);
```

Comment:
same comment as in timeline-chart/src/layer/time-graph-chart-selection-range.ts about APIs.

Reply:
This should also be resolved.